src/lib.rs

```diff
@@ -104,13 +105,22 @@ const BUCKETS: usize = (POINTER_WIDTH + 1) as usize;
 pub struct ThreadLocal<T: Send> {
     /// The buckets in the thread local. The nth bucket contains `2^(n-1)`
     /// elements. Each bucket is lazily allocated.
-    buckets: [AtomicPtr<UnsafeCell<Option<T>>>; BUCKETS],
+    buckets: [AtomicPtr<Entry<T>>; BUCKETS],
+
+    /// The number of values in the thread local. This can be less than the real number of values,
+    /// but is never more.
+    values: AtomicUsize,
 
     /// Lock used to guard against concurrent modifications. This is taken when
     /// there is a possibility of allocating a new bucket, which only occurs
-    /// when inserting values. This also guards the counter for the total number
-    /// of values in the thread local.
-    lock: Mutex<usize>,
+    /// when inserting values.
+    lock: Mutex<()>,
+}
+
+struct Entry<T> {
+    present: AtomicBool,
+    // Use MaybeUninit once the MSRV has been bumped.
```

Comment:
Might as well just bump the MSRV.

Reply:
The problem is that this crate is a dependency of Regex and it'd be a shame if they had to pin their version of this crate since I don't think they want to bump their MSRV.